### PR TITLE
Added project controller

### PR DESCRIPTION
**Pull request by ValentynSharshon**

### CODE REVIEW

File: src/main/java/app/controller/ProjectController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Project management", description = "Endpoints for managing projects")
+@RestController
+@RequestMapping("/projects")
```
Comment (makaroshyna) [inline src/main/java/app/controller/ProjectController.java:28]: Don't forget to change it to /api/projects

File: src/main/java/app/controller/ProjectController.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@RequiredArgsConstructor
+public class ProjectController {
+    private final ProjectService projectService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
```
Comment (makaroshyna) [inline src/main/java/app/controller/ProjectController.java:34]: Add swagger Operation annotation to controller methods.

File: src/main/java/app/controller/ProjectController.java (hunk 1 above)
Comment (EduarfF) [inline src/main/java/app/controller/ProjectController.java:28]: /api/projects

File: src/main/java/app/dto/project/ProjectRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import jakarta.validation.constraints.FutureOrPresent;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import java.time.LocalDate;
+
+public record ProjectRequestDto(
+        @NotBlank
```
Comment (makaroshyna) [inline src/main/java/app/dto/project/ProjectRequestDto.java:11]: You can add messages here, like this
```suggestion
        @NotBlank(message = "Project name must be defined")
```

**Review by EduarfF — CHANGES_REQUESTED** [review]

File: src/main/java/app/dto/project/ProjectRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        @Size(max = 250)
+        String description,
+        @NotNull
+        @FutureOrPresent
+        @JsonFormat(pattern = "dd-MM-yyyy")
+        LocalDate startDate,
+        @NotNull
+        @FutureOrPresent
+        @JsonFormat(pattern = "dd-MM-yyyy")
```
Comment (makaroshyna) [inline src/main/java/app/dto/project/ProjectRequestDto.java:23]: The format for date in our project is already "YYYY-MM-dd", so maybe change it to that to be more uniform?

**Review by makaroshyna — CHANGES_REQUESTED** [review]
Good! Left some minor comments :)

**Review by makaroshyna — APPROVED** [review]

**Review by EduarfF — APPROVED** [review]

**Pull request merged**